### ISSUE TITLE
ViewRenderable size is limited by screen size

### ISSUE THREAD

**Issue by yperess** [google-ar/sceneform-android-sdk#358]
We're considering switching over to sceneform for our AR rendering but we're running into some issues with ViewRenderable. We've created a large TextView that we want to display, the layout parameters are set to `wrap_content` in both dimensions. Even though the view isn't attached to a parent (in our hierarchy) `onMeasure` is called for the TextView with the dimensions of the `ArSceneView` and mode `MeasureSpec.AT_MOST` which causes the text to wrap (I believe triggered from RenderViewToExternalTexture).

Expected behavior: The view should be able to be unbounded by the screen size. Small warning: in our own VirtualView we've noticed that textures must be capped at 2MB. So measurement of the view should be done then the underlying surface texture might need to be scaled down.

**Comment by romainguy** [comment]
Textures don't have a size cap in number of bytes, they will however be limited by the capabilities of the underlying GPU driver (there's a max texture size). This means they cannot be completely unconstrained. There's also the fact that in many cases being unbounded for text won't necessarily be what you want. We should probably have a way for view renderables to choose between a "comfortable" mode (current behavior) and a "max size" mode (using AT_MOST and the max possible texture size).

**Comment by yperess** [comment]
Good catch, I don't know why I didn't think to look at the OpenGL APIs for the texture sizing. 2MB came from testing a bunch of devices. I've updated our own renderer to perform the following:

1. Get the max texture size:

```
val maxTextureSize = IntArray(1).let {
    GLES10.glGetIntegerv(GL10.GL_MAX_TEXTURE_SIZE, it, 0)
    it[0]
}
```

2. On each measure pass allow the view to measure freely, but if the view's layout params specify `wrap_content` or `match_parent` (treated as `wrap_content` in AR/VR) then we check if `width * height > maxTextureSize`. If so, we keep the aspect ratio and set the buffer size such that it doesn't exceed the texture size.

Regardless, I think the key here is deciding how `wrap_content` and `match_parent` behave in 3D. In my personal opinion (and I welcome any differing views here) both should be treated as unbound and allow the view to measure itself (unless a max width/height is also specified).
